tools/wptserve/wptserve/request.py

```diff
@@ -610,8 +607,33 @@ def from_field_storage(cls, fs):
         return self
 
 
+class BinaryCookieParser(BaseCookie):
+    """A subclass of BaseCookie that returns values in binary strings
+
+    This is not intended to store the cookies; use Cookies instead."""
+
+    def value_decode(self, val):
+        """Decode value from network to (real_value, coded_value).
+
+        Override BaseCookie.value_decode."""
+        real = val.encode('iso-8859-1') if isinstance(val, text_type) else val
```

Comment:
Can we use `isomorphic_encode` from the other PR?

Reply:
Done. I hadn't rebased this branch.